### PR TITLE
Add a splash screen when Amaze loads

### PR DESCRIPTION
**Pull request by dhruvpatidar359**
Signed-off-by: dhruv [dhruvpatidar35@gmail.com](mailto:dhruvpatidar35@gmail.com)
closes https://github.com/TeamAmaze/AmazeFileManager/issues/3139

 Description

Issue tracker   
Fixes https://github.com/TeamAmaze/AmazeFileManager/issues/3139

Automatic tests

- [ ] Added test cases
  
 Manual tests
- [x] Done  
  

 Build tasks success  

Successfully running following tasks on local:
- [ ] `./gradlew assembledebug`
- [ ] `./gradlew spotlessCheck`

### CODE REVIEW

**Comment by dhruvpatidar359** [comment]
hello @EmmanuelMess   can you please review this PR .

**Comment by dhruvpatidar359** [comment]
@EmmanuelMess  may you please help me how i can i resolve these checks , as am not getting a proper solution for it .

File: app/build.gradle [hunk 1]
```diff
@@ -120,6 +120,7 @@ dependencies {
     implementation "com.google.android.material:material:$androidMaterialVersion"
     implementation "androidx.fragment:fragment-ktx:$androidXFragmentVersion"
     implementation "androidx.appcompat:appcompat:$androidXAppCompatVersion"
+    implementation 'androidx.core:core-splashscreen:1.0.0-beta01'
```
Comment (EmmanuelMess) [inline app/build.gradle:123]: You might not want to use the splash screen library if it has API level issues.

**Review by EmmanuelMess — CHANGES_REQUESTED** [review]
The image used should not be a png, it should be an svg.

**Comment by EmmanuelMess** [comment]
You should run spotlessApply to fix check spotless.

**Comment by EmmanuelMess** [comment]
```

com.amaze.filemanager.asynchronous.asynctasks.SshDeleteTaskTest > testDeleteSshFileAccessDenied[16] FAILED
    java.lang.NoSuchMethodError: 'android.graphics.drawable.Drawable android.content.res.Resources$Theme.getDrawable(int)'
        at androidx.core.splashscreen.SplashScreen$Impl.install(SplashScreen.kt:264)
        at androidx.core.splashscreen.SplashScreen.install(SplashScreen.kt:189)
        at androidx.core.splashscreen.SplashScreen.access$install(SplashScreen.kt:121)
        at androidx.core.splashscreen.SplashScreen$Companion.installSplashScreen(SplashScreen.kt:146)
        at androidx.core.splashscreen.SplashScreen.installSplashScreen(SplashScreen.kt)
        at com.amaze.filemanager.ui.activities.MainActivity.onCreate(MainActivity.java:331)
        at android.app.Activity.performCreate(Activity.java:5008)
        at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1079)
        at org.robolectric.android.internal.RoboMonitoringInstrumentation.callActivityOnCreate(RoboMonitoringInstrumentation.java:284)
        at org.robolectric.android.controller.ActivityController.lambda$create$0(ActivityController.java:113)
        at org.robolectric.shadows.ShadowPausedLooper.runPaused(ShadowPausedLooper.java:203)
        at org.robolectric.android.controller.ActivityController.create(ActivityController.java:113)
        at org.robolectric.android.controller.ActivityController.create(ActivityController.java:118)
        at org.robolectric.android.internal.RoboMonitoringInstrumentation.startActivitySyncInternal(RoboMonitoringInstrumentation.java:103)
        at org.robolectric.android.internal.LocalActivityInvoker.startActivity(LocalActivityInvoker.java:33)
        at org.robolectric.android.internal.LocalActivityInvoker.startActivity(LocalActivityInvoker.java:38)
        at androidx.test.core.app.ActivityScenario.launchInternal(ActivityScenario.java:265)
        at androidx.test.core.app.ActivityScenario.launch(ActivityScenario.java:195)
        at com.amaze.filemanager.asynchronous.asynctasks.AbstractDeleteTaskTestBase.doTestDeleteFileAccessDenied(AbstractDeleteTaskTestBase.kt:107)
        at com.amaze.filemanager.asynchronous.asynctasks.SshDeleteTaskTest.testDeleteSshFileAccessDenied(SshDeleteTaskTest.kt:53)

```

Are you using getDrawable on the wrong API level?

**Comment by dhruvpatidar359** [comment]
@EmmanuelMess  ok I will remove the library.
May you please explain the meaning of making the splash screen by hand as you said earlier

**Comment by EmmanuelMess** [comment]
> May you please explain the meaning of making the splash screen by hand as you said earlier

I think you can use an activity to show an image, and start the main activity from there, so that the main activity loads with the initial image as background.

**Comment by dhruvpatidar359** [comment]
@EmmanuelMess  , i am not able to create a new activity as the minsdk is 14 but requires 16(minimum) for that i have to chaange the minsdk 14 to 16
image

**Comment by EmmanuelMess** [comment]
> @EmmanuelMess , i am not able to create a new activity as the minsdk is 14 but requires 16(minimum) for that i have to change the minsdk 14 to 16 image

That's with the dialog, you should create it by hand, or convert what the dialog creates to an API 14 compatible version.

**Comment by dhruvpatidar359** [comment]
@EmmanuelMess  is there any svg available for the icon in the project as i am not able to find it ?

**Pull request closed without merging**